packages/babel-plugin-lingui-macro/src/index.ts

```diff
@@ -46,6 +47,19 @@ function reportUnsupportedSyntax(path: NodePath, e: Error) {
   throw codeFrameError
 }
 
+function shouldStripMessageProp(opts: LinguiPluginOpts) {
+  if (opts.extract) {
+    // never strip message during extraction process
+    return false
+  }
```

Comment:
i think this is not needed, because this case is not possible. The babel options which extractor uses are not exposed to the userland and could not be overridden by users. So i could not imagine the situation when `{extract: true,  stripMessageProp: true}` would be passed together.  

This will allow to delete some of the tests, less code, less things to maintain. 

Thanks!

Reply:
Well spotted! I added it in the initial implementation based on v4. In that version when I ran the extract command the message was stripped if I had `stripMessageProp: true` set in the Babel config unless I had this line to explicitly return `false` during extraction.

I tested it now against the new v5 implementation and it indeed seems like the `stripMessageField: true` option does not affect extract. However if I remove this check, some of the existing test cases will break. For example this one since it will remove the message prop due to `NODE_ENV: "production"`: https://github.com/lingui/js-lingui/blob/3a513504784802257fb5ba86b6814f0ea08b55f8/packages/babel-plugin-lingui-macro/test/js-t.test.ts#L208-L222

How do you think it should be handled?

One alternative that I see could be to change the implementation to something like:

```javascript
function shouldStripMessageProp(opts: LinguiPluginOpts) {
  if (typeof opts.stripMessageField === "boolean") {
    // if explicitly set in options, use it
    return opts.stripMessageField
  }
  // default to strip message in production if no explicit option is set and not during extract
  return process.env.NODE_ENV === "production" && !opts.extract
```

This way it would be more aligned with the `stripNonEssentialProps` value and won't mess up extract if someone runs extract with NODE_ENV set to "production".

Reply:
> Well spotted! I added it in the initial implementation based on v4. In that version when I ran the extract command the message was stripped if I had stripMessageProp: true set in the Babel config unless I had this line to explicitly return false during extraction.

That's strange, because nothing was changed between 4 and 5 in that regard. Lingui extractor does not use user's babel config (intentionally) and no options from there could affect extraction process. 

Your proposed alternative is fine.

Reply:
It might have been me that messed something up when I tested it before, as I was testing a few different implementations in v4. Anyway I confirmed in v5 that it doesn't seem to be a possible case like you said. I have changed the implementation now to what I proposed in my previous comment since that doesn't break any of the existing test cases.